src/backend/indirectrange.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include "../ds/ptrwrap.h"
+
+namespace snmalloc
+{
+  template<typename T>
+  static T& IndirectRange_default_kparent(T* x)
+  {
+    return *x;
+  }
+
+  template<
+    typename ParentRange,
+    typename KArg_ = typename ParentRange::State*,
```

Comment:
Could the following be made to work:
```suggestion
    typename KArg_ = typename ParentRange::State&,
```
Then it would ensure that callers had to provide one?